tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Http/WebRequest/HttpWebRequest_EndGetResponse_Integration.cs

```diff
@@ -68,12 +68,15 @@ internal static CallTargetReturn<TReturn> OnMethodEnd<TTarget, TReturn>(TTarget
                 // Since it is possible for users to manually propagate headers (which we should
                 // overwrite), check our cache which will be populated with header objects
                 // that we have injected context into
-                SpanContext existingSpanContext = null;
+                PropagationContext existingContext = default;
                 if (HeadersInjectedCache.TryGetInjectedHeaders(request.Headers))
                 {
-                    existingSpanContext = SpanContextPropagator.Instance.Extract(request.Headers.Wrap());
+                    existingContext = SpanContextPropagator.Instance.Extract(request.Headers.Wrap());
+                    Baggage.Current.Merge(existingContext.Baggage);
```

Comment:
In this case I wonder if we should completely overwrite the AsyncLocal Baggage instead of merge. I'm not 100% sure which way we should go here

Reply:
Same 😅 

My thinking was if we have different baggage keys, might as well include them all (shouldn't hurt?). But there is the issue of having conflicting baggage items with same key and different values.

You are more familiar with the special cases in these integrations, so I'm happy to follow your suggestion.

Reply:
Actually I'm changing my tune, I don't think we should set the current baggage. We've already propagated baggage through the HTTP headers at this point, and when this method is called this is presumably the "bottom" of the call chain, and it may have been called on an entirely different thread.

Reply:
So I should remove `Baggage.Current.Merge(existingContext.Baggage)` here?

Reply:
Yes, let's remove `Baggage.Current.Merge(existingContext.Baggage)` here

Reply:
Stopped merging the extracted baggage and added a comment (mostly quoting what you said above).